zkevm-circuits/src/evm_circuit/execution/begin_tx.rs

```diff
@@ -71,47 +79,49 @@ impl<F: Field> ExecutionGadget<F> for BeginTxGadget<F> {
         // Use rw_counter of the step which triggers next call as its call_id.
         let call_id = cb.curr.state.rw_counter.clone();
 
-        let tx_id = cb.query_cell();
-        cb.call_context_lookup(
-            1.expr(),
+        let tx_id = cb.query_u64();
+        cb.call_context_lookup_write_unchecked(
             Some(call_id.expr()),
             CallContextFieldTag::TxId,
-            tx_id.expr(),
+            tx_id.to_word(),
         ); // rwc_delta += 1
         let mut reversion_info = cb.reversion_info_write(None); // rwc_delta += 2
-        cb.call_context_lookup(
-            1.expr(),
+        cb.call_context_lookup_write_unchecked(
             Some(call_id.expr()),
             CallContextFieldTag::IsSuccess,
-            reversion_info.is_persistent(),
+            Word::from_lo_unchecked(reversion_info.is_persistent()),
         ); // rwc_delta += 1
 
-        let [tx_nonce, tx_gas, tx_caller_address, tx_callee_address, tx_is_create, tx_call_data_length, tx_call_data_gas_cost] =
-            [
-                TxContextFieldTag::Nonce,
-                TxContextFieldTag::Gas,
-                TxContextFieldTag::CallerAddress,
-                TxContextFieldTag::CalleeAddress,
-                TxContextFieldTag::IsCreate,
-                TxContextFieldTag::CallDataLength,
-                TxContextFieldTag::CallDataGasCost,
-            ]
-            .map(|field_tag| cb.tx_context(tx_id.expr(), field_tag, None));
-        let tx_caller_address_is_zero = IsZeroGadget::construct(cb, tx_caller_address.expr());
+        let [tx_nonce, tx_gas, tx_is_create, tx_call_data_length, tx_call_data_gas_cost] = [
+            TxContextFieldTag::Nonce,
+            TxContextFieldTag::Gas,
+            TxContextFieldTag::IsCreate,
+            TxContextFieldTag::CallDataLength,
+            TxContextFieldTag::CallDataGasCost,
+        ]
+        .map(|field_tag| cb.tx_context(tx_id.expr(), field_tag, None));
+        let [tx_gas_price, tx_value] = [TxContextFieldTag::GasPrice, TxContextFieldTag::Value]
```

Comment:
Here is the comparison for main branch/evm word-lo-hi branch utilization

### Storage_1
```
Main:
+-----------------------------------+------------------------------+-------------------------+
| "storage_1" total_available_cells | "storage_1" total_used_cells | "storage_1" Utilization |
+-----------------------------------+------------------------------+-------------------------+
| 25480                             | 6482                         | 25.4                    |
+-----------------------------------+------------------------------+-------------------------+

+-----------------------------------+------------------------------+-------------------------+
| "storage_1" total_available_cells | "storage_1" total_used_cells | "storage_1" Utilization |
+-----------------------------------+------------------------------+-------------------------+
| 24220                             | 7273                         | 30.0                    |
+-----------------------------------+------------------------------+-------------------------+
```

### Storage_2
```
Main
+-----------------------------------+------------------------------+-------------------------+
| "storage_2" total_available_cells | "storage_2" total_used_cells | "storage_2" Utilization |
+-----------------------------------+------------------------------+-------------------------+
| 1456                              | 467                          | 32.1                    |
+-----------------------------------+------------------------------+-------------------------+

+-----------------------------------+------------------------------+-------------------------+
| "storage_2" total_available_cells | "storage_2" total_used_cells | "storage_2" Utilization |
+-----------------------------------+------------------------------+-------------------------+
| 1384                              | 14                           | 1.0                     |
+-----------------------------------+------------------------------+-------------------------+
```

### byte_lookup
```
Main 
+-------------------------------------+--------------------------------+---------------------------+
| "byte_lookup" total_available_cells | "byte_lookup" total_used_cells | "byte_lookup" Utilization |
+-------------------------------------+--------------------------------+---------------------------+
| 8736                                | 6786                           | 77.7                      |
+-------------------------------------+--------------------------------+---------------------------+

+-------------------------------------+--------------------------------+---------------------------+
| "byte_lookup" total_available_cells | "byte_lookup" total_used_cells | "byte_lookup" Utilization |
+-------------------------------------+--------------------------------+---------------------------+
| 8304                                | 6606                           | 79.6                      |
+-------------------------------------+--------------------------------+---------------------------+

```

Overall before/after utilization look not significant diff, some even decrease. After u16 optimisation it will be optimise more.

Reply:
> TransferWithGasFeeGadget

E.g. `MulWordByU64Gadget` there is problematics design which make the refactor harder. The thing is, `multiplicand` is passed from caller, however assign inside `MulWordByU64Gadget`. So, `multiplicand` can not took generics type, otherwise it make `assign` harder, because for now we need to know exact type when we doing assigment. 

I would suggest to refactor cell usage gradually maybe later as an optimisation. For now we took functionality correct first :P

Reply:
Sounds good, we can leave this for a future optimization :)  Thanks for the utilization results!  They are very useful because they show that the cell usage hasn't grown!